waspc/waspls/src/Wasp/LSP/Syntax.hs

```diff
@@ -27,21 +31,39 @@ lspPositionToOffset srcString (J.Position l c) =
 
 -- | Move to the node containing the offset.
 --
--- This tries to prefer non-trivia tokens where possible. If the offset falls
--- exactly between two tokens, it choses the left-most non-trivia token.
-toOffset :: Int -> Traversal -> Traversal
-toOffset targetOffset start = go $ bottom start
+-- If the offset falls on the border between two nodes, it tries to first choose
+-- the leftmost non-trivia token, and then the leftmost token.
+locationAtOffset :: Int -> Traversal -> Traversal
+locationAtOffset targetOffset start = go $ bottom start
   where
     go :: Traversal -> Traversal
     go at
       | offsetAt at == targetOffset = at
       | offsetAfter at > targetOffset = at
-      | offsetAfter at == targetOffset && not (S.syntaxKindIsTrivia (kindAt at)) =
-          at
+      | offsetAfter at == targetOffset =
+          if not $ S.syntaxKindIsTrivia $ kindAt at
+            then at
+            else case at & next of
+              Just at' | not (S.syntaxKindIsTrivia (kindAt at')) -> at'
+              _ -> at
       -- If @at & next@ fails, the input doesn't contain the offset, so just
       -- return the last node instead.
       | otherwise = maybe at go $ at & next
 
+-- | Check if all of the supplied predicates are true
+allP :: Foldable f => f (a -> Bool) -> a -> Bool
+allP preds x = all ($ x) preds
+
+-- | Check if any of the supplied predicates are true
+anyP :: Foldable f => f (a -> Bool) -> a -> Bool
+anyP preds x = any ($ x) preds
```

Comment:
Maybe move these functions to some Util module? They are in no way special for this module, right?